### PR TITLE
Add Razor (cshtml) highlighting

### PR DESCRIPTION
**Pull request by worldbeater**
## Description

Currently, Github won't highlight `.cshtml` files properly. See https://github.com/github/linguist/issues/3760. Let's use highlighting from https://github.com/austincummings/razor-plus

Razor is like Django - a templating engine based on HTML and C#. See [MSDN documentation](https://docs.microsoft.com/en-us/aspnet/core/razor-pages/?view=aspnetcore-2.1&tabs=visual-studio) for more info. Now, GitHub treats `.cshtml` files as regular C# files, but this is not quite right. If we use the grammar @austincummings made, we'll get a much better syntax highlighting.

## Checklist:

- [x] **I am changing the source of a syntax highlighting grammar**
  - [Old grammar](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgithub.com%2Fatom%2Flanguage-csharp%2Fblob%2Fmaster%2Fgrammars%2Fcsharp.cson&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Faspnet%2FMusicStore%2Fblob%2Fmaster%2Fsamples%2FMusicStore%2FViews%2FStore%2FBrowse.cshtml&code=)
  - [New grammar](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Faustincummings%2Frazor-plus%2Fmaster%2Fsyntaxes%2Fcshtml.tmLanguage.json&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Faspnet%2FMusicStore%2Fblob%2Fmaster%2Fsamples%2FMusicStore%2FViews%2FStore%2FBrowse.cshtml&code=)
  - [x] The extension of the language is used in hundreds of repositories on GitHub.com.
    - [Search results for `.cshtml`](https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3Acshtml+NOT+nothack)
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [MusicStore](https://github.com/aspnet/MusicStore/blob/master/samples/MusicStore/Views/Store/Browse.cshtml)

### CODE REVIEW

**Comment by pchaigno** [comment]
Please fill the template. I have the context, but other reviewers may not. IMO, the best match is *I am changing the source of a syntax highlighting grammar*.

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -1866,6 +1865,16 @@ HTML+PHP:
...
   language_id: 151
+HTML+Razor:
+  type: markup
+  tm_scope: text.html.razor
+  group: HTML
+  extensions:
+  - ".cshtml"
+  ace_mode: razor
+  codemirror_mode: razor
+  codemirror_mime_type: text/x-razor
```
[the remark this answers is not in the document]
Comment (worldbeater) [reply lib/linguist/languages.yml]: Actually I didn't. Could you please explain what are these options responsible for? Is it something like a "fallback value" or something else etc.

**Comment by worldbeater** [comment]
@pchaigno thanks for reviewing this! CI checks passed 👌

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -1866,6 +1865,16 @@ HTML+PHP:
...
+HTML+Razor:
+  type: markup
+  tm_scope: text.html.razor
+  group: HTML
+  extensions:
+  - ".cshtml"
+  ace_mode: razor
+  codemirror_mode: htmlmixed
+  codemirror_mime_type: text/html
+  language_id: 432
```
Comment (lildude) [inline lib/linguist/languages.yml]: 🤔 This appears to have been manually set. Please use `script/update-ids` as detailed in CONTRIBUTING.md to set the language ID. You'll need to remove `432` from your local file before running the script in order for it to be replaced.

Comment (pchaigno) [reply lib/linguist/languages.yml]: Wow! We should probably detect that in tests.

Comment (lildude) [reply lib/linguist/languages.yml]: Yup. Coming up.

Comment (worldbeater) [reply lib/linguist/languages.yml]: Sorry for that. I was experiencing mysterious issues when trying to follow the manual at CONTRIBUTING.md. `byebug 10.0.2` is throwing exceptions:
```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /home/artyom/Github/linguist/vendor/gems/ruby/2.5.0/gems/byebug-10.0.2/ext/byebug
/usr/bin/ruby2.5 -r ./siteconf20181107-23957-1cf8lyo.rb extconf.rb
mkmf.rb can't find header files for ruby at /usr/lib/ruby/include/ruby.h
```

Comment (pchaigno) [reply lib/linguist/languages.yml]: @worldbeater You need the ruby dev package, `ruby-dev` under Ubuntu.

**Pull request merged**